sdk/rust/oak_tests/src/lib.rs

```diff
@@ -38,33 +38,42 @@ pub enum Profile {
 
 // TODO(#544): re-enable unit tests of SDK functionality
 
-/// Uses cargo to compile a Rust manifest to Wasm bytes.
-pub fn compile_rust_wasm(
-    manifest_path: &str,
-    module_wasm_file_name: &str,
-    profile: Profile,
-) -> anyhow::Result<Vec<u8>> {
-    // Use a fixed target directory, because `--target-dir` influences SHA256 hash of Wasm module.
-    // Directory should end with `target` so that it gets automatically ignored by our `.gitignore`.
-    let mut target_dir = PathBuf::from(manifest_path);
-    target_dir.pop();
-    target_dir.push("target");
+/// Returns the path to the Wasm file produced by compiling the provided `Cargo.toml` file.
+fn build_wasm_module_path(manifest_path: &str, profile: Profile) -> String {
+    let metadata = cargo_metadata::MetadataCommand::new()
+        .manifest_path(manifest_path)
+        .exec()
+        .unwrap();
+    let package_name = &metadata.root_package().unwrap().name;
+    let mut module_path = PathBuf::from(&metadata.target_directory);
+    module_path.pop();
+    module_path.push(format!(
+        "target/wasm32-unknown-unknown/{}/{}.wasm",
+        match profile {
+            Profile::Release => "release",
+            Profile::Debug => "debug",
+        },
+        package_name
+    ));
+    module_path
+        .to_str()
+        .expect("could not convert PathBuf to String")
+        .to_string()
+}
 
+/// Uses cargo to compile a Rust manifest to Wasm bytes.
+pub fn compile_rust_wasm(manifest_path: &str, profile: Profile) -> anyhow::Result<Vec<u8>> {
```

Comment:
Does the function with the same name in `oak_functions` need to be updated too? Perhaps in a separate PR, or as part of #1965.

Reply:
Yes, I think all of them should be unified in #1965.